packages/http-service/src/stream.js

```diff
@@ -0,0 +1,27 @@
+import Rx from "rxjs";
+
+import ConnectionManager from "connection-manager";
+import { XHRConnection, ConnectionEvent } from "connections";
+
+export default function stream(url, options = {}) {
+  return Rx.Observable.create(function(observer) {
+    const connection = new XHRConnection(url, options);
+
+    connection.addListener(ConnectionEvent.DATA, function(event) {
+      observer.next(event.target.response);
+    });
+    connection.addListener(ConnectionEvent.ERROR, function(event) {
+      observer.error(event);
+    });
+    connection.addListener(ConnectionEvent.COMPLETE, function() {
+      observer.complete();
+    });
+
+    ConnectionManager.enqueue(connection);
+
+    // Closing the connection when there are no subscribers left
+    return function teardown() {
+      connection.close();
```

Comment:
Let's use the connection manager `dequeue` instead of directly interacting with the connection.

Reply:
Makes sense